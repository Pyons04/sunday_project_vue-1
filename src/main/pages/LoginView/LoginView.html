<div>
    <table class="border-separate p-5 px-10">
        <thead>
            <tr>
                <th class="w-24"></th>
                <th></th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td class="">ユーザID：</td>
                <td class="bg-blue-200"><input type="text" id="username" v-model="username"></td>
            </tr>
            <tr>
                <td class="">Password：</td>
                <td class="bg-blue-200"><input type="password" id="password" v-model="password"></td>
            </tr>
        </tbody>
    </table>
    <div class="p-5 px-10">
        <button class="bg-blue-200 p-2 w-24 rounded-xl" @click="login">ログイン</button>
    </div>
</div>
