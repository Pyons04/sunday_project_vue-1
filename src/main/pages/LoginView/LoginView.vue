<template src="./LoginView.html"></template>

<script>
import axios from 'axios';
// import { ref } from 'vue';

export default{
    name: 'LoginView',
    data() {
        return {
            username: '',
            password: '',
            msg: ''
        }
    },
    methods: {
        login: function() {
            console.log('login')
            const data = {
                username: this.username,
                password: this.password
            }
            console.log(this.$store.state.loggedIn);
            axios.post('/ticket/api/login/', data)
            .then(
                response => { console.log(response);
                this.$store.commit('login',{state: true})
                console.log('redirect path: '+this.$route.query.redirect);
                console.log('login state: '+this.$store.state.loggedIn);
                if(this.$route.query.redirect == undefined){
                    this.$router.push('/list')
                }
                else{
                    this.$router.push(this.$route.query.redirect);
                }
            })
            .catch((err) => {
                this.msg = err
                console.log(err);
            });
        },
        onKeyDown(event) {
            const key = event.keyCode;
            // console.log(key);
            if(key == 13){
                // console.log('Enter');
                // console.log(this.username);
                // console.log(this.password);
                if(this.username == ''){
                    console.log('enmpty username');
                    this.$refs.username.focus();
                }
                else if(this.password == ''){
                    console.log('enmpty password');
                    this.$refs.password.focus();
                }
                else if(this.username != '' && this.password != ''){
                    // console.log('exe login');
                    this.login();
                }
                
            }
        }
    }
}
</script>