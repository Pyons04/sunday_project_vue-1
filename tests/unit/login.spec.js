import { mount, flushPromises } from '@vue/test-utils'
import LoginView from '../../src/main/pages/LoginView/LoginView.vue'

describe('SnapShot testing for login view.', () => {
  it('SnapShot LoginView.', () => {
    const wrapper = mount(LoginView)
    //expect(wrapper.element).toMatchSnapshot();
  }) 
})

test('SnapShot LoginView.', () => {
  const wrapper = mount(LoginView)
  const input_form = wrapper.find('tbody').findAll('tr')
  expect(input_form[0].findAll('td')[0].text()).toBe('ユーザID：');
  expect(input_form[0].findAll('td')[1].find('input').attributes('id')).toBe('username');
  expect(input_form[1].findAll('td')[0].text()).toBe('Password：');
  expect(input_form[1].findAll('td')[1].find('input').attributes('id')).toBe('password');

  expect(wrapper.find('button').text()).toBe('ログイン')
}) 