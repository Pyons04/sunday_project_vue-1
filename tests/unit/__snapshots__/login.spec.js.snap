// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot testing for login view. SnapShot LoginView. 1`] = `
<div>
  <table
    class="border-separate p-5 px-10"
  >
    <thead>
      <tr>
        <th
          class="w-24"
        />
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class=""
        >
          ユーザID：
        </td>
        <td
          class="bg-blue-200"
        >
          <input
            id="username"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td
          class=""
        >
          パスワード：
        </td>
        <td
          class="bg-blue-200"
        >
          <input
            id="password"
            type="password"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="p-5 px-10"
  >
    <button
      class="bg-blue-200 p-2 w-24 rounded-xl"
    >
      ログイン
    </button>
  </div>
</div>
`;
